src/ops/poseidon_hash.rs

```diff
@@ -0,0 +1,60 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use super::AbiCall;
+use crate::call_context::{CallContext, Resolver};
+use crate::VMError;
+
+use canonical::Store;
+use wasmi::{RuntimeArgs, RuntimeValue, ValueType};
+
+use dusk_bls12_381::BlsScalar;
+use dusk_bytes::{DeserializableSlice, Serializable};
+use poseidon252::sponge::hash;
+
+pub struct PoseidonHash;
+
+impl<E: Resolver<S>, S: Store> AbiCall<E, S> for PoseidonHash {
+    const ARGUMENTS: &'static [ValueType] =
+        &[ValueType::I32, ValueType::I32, ValueType::I32];
+    const RETURN: Option<ValueType> = None;
+
+    fn call(
+        context: &mut CallContext<E, S>,
+        args: RuntimeArgs,
+    ) -> Result<Option<RuntimeValue>, VMError<S>> {
+        if let &[RuntimeValue::I32(ret_addr), RuntimeValue::I32(ofs), RuntimeValue::I32(len)] =
+            args.as_ref()
+        {
+            let ofs = ofs as usize;
+            let len = len as usize;
```

Comment:
I would encode the len inside the BlsScalar slice. I do something similar with the public inputs:
```rust
let pub_inp_len = mem[pub_inp_addr] as usize;
    // Build Public Inputs vector
    let mut pi_bytes = vec![0u8; pub_inp_len * PublicInput::serialized_size()];
    pi_bytes.copy_from_slice(
        &mem[pub_inp_addr + 1
            ..pub_inp_addr
                + 1
                + (pub_inp_len * PublicInput::serialized_size())],
    );
    pi_bytes[..]
        .chunks(PublicInput::serialized_size())
        .map(|chunk| {
            PublicInput::from_bytes(chunk)
                .map_err(|_| VMError::ABICallExecError)
        })
        .collect::<Result<Vec<PublicInput>, VMError<S>>>()
```

Reply:
Mhm… I'm failing to see the advantage in doing so, since it's transparent to the user and add a complexity on our end (the user will never pass the `len` manually, just a vector of `BlsScalar`).